models/correlospinmatrix_detector.h

```diff
@@ -0,0 +1,309 @@
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef CORRELOSPINMATRIX_DETECTOR_H
+#define CORRELOSPINMATRIX_DETECTOR_H
+
+
+#include <vector>
+#include <deque>
+#include "nest.h"
+#include "event.h"
+#include "node.h"
+#include "pseudo_recording_device.h"
+
+/* BeginDocumentation
+
+   Name: correlospinmatrix_detector - Device for measuring the covariance matrix from several inputs
+
+   Description: The correlospinmatrix_detector is a recording device. It is used to record
+   correlations
+   from binary neurons from several binary sources and calculates the raw auto and cross correlation
+   binned to bins of duration delta_tau.
+   The result can be obtained via GetStatus under the key /count_covariance.
+   The result is a tensor of rank 3 of size N_channels x N_channels, with each entry C_ij being a
+   vector of
+   size 2*tau_max/delta_tau + 1 containing the histogram for the different time lags.
+
+   The bins are centered around the time difference they represent, and are left-closed
+   and right-open in the lower triangular part of the matrix. On the diagonal and in the upper
+   triangular part the intervals are left-open and right-closed. This ensures proper counting
+   of events at the border of bins.
+
+   The correlospinmatrix_detector has a variable number of inputs which can be set via SetStatus
+   under the key N_channels. All incoming connections to a specified receptor will be pooled.
+
+   Parameters:
+   Tstart     double    - Time when to start counting events. This time should be set to at least
+                          start + tau_max in order to avoid edge effects of the correlation counts.
+   Tstop      double    - Time when to stop counting events. This time should be set to at most
+                          Tsim - tau_max, where Tsim is the duration of simulation,
+                          in order to avoid edge effects of the correlation counts.
+   delta_tau  double    - bin width in ms. This has to be an odd multiple of the resolution, to
+   allow
+                          the symmetry between positive and negative time-lags.
+   tau_max    double    - one-sided width in ms. In the lower triangular part events with
+   differences in
+                          [0, tau_max+delta_tau/2) are counted. On the diagonal and in
+                          the upper triangular part events with differences in (0,
+   tau_max+delta_tau/2]
+
+   N_channels long      - The number of inputs to correlate. This defines the range of
+   receptor_type. Default is
+   1.
+                          Setting N_channels clears count_covariance, covariance and n_events.
```

Comment:
you are right, setting the of the reproperties also clears the measurements; updated doc.